src/rgw/rgw_lua.cc

```diff
@@ -238,6 +238,9 @@ int install_packages(const DoutPrefixProvider *dpp, rgw::sal::Driver* driver,
     ldpp_dout(dpp, 1) << "Lua ERROR: failed to create temporary directory from template: " << 
       tmp_path_template << ". error: " << rc << dendl;
     return rc;
+  } else {
```

Comment:
making the direcory works ok. it is the call in line 268 is failing.
also, `luarocks` needs write access to that directory

Reply:
the luarocks install is also working fine as it is still running as user root:

```
tree /tmp/rgw_luarocks/0SejpX/
/tmp/rgw_luarocks/0SejpX/
|-- lib
|   |-- lua
|   |   `-- 5.3
|   |       |-- mime
|   |       |   `-- core.so
|   |       `-- socket
|   |           |-- core.so
|   |           |-- serial.so
|   |           `-- unix.so
|   `-- luarocks
```

but as soon as the rgw drops to the user `ceph` in normal operration access to the luarocks files is no longer possible